ownCloud/Client/Sharing/PublicLinkTableViewController.swift

```diff
@@ -0,0 +1,273 @@
+ *
+ */
+
+import UIKit
+import ownCloudSDK
+
+class PublicLinkTableViewController: SharingTableViewController {
+
+	// MARK: - Instance Variables
+	var shares : [OCShare] = []
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		messageView = MessageView(add: self.view)
+
+		self.navigationItem.title = "Links".localized
+		self.navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(dismissAnimated))
+		self.navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .add, target: self, action: #selector(addPublicLink))
+
+		addHeaderView()
+		addPrivateLinkSection()
+
+		shareQuery = core?.sharesWithReshares(for: item, initialPopulationHandler: { [weak self] (sharesWithReshares) in
+			if let self = self, sharesWithReshares.count > 0 {
+				self.shares = sharesWithReshares.filter { (share) -> Bool in
+					return share.type == .link
+				}
+				OnMainThread {
+					self.addShareSections()
+				}
+			}
+		}, changesAvailableNotificationHandler: { [weak self] (sharesWithReshares) in
+			guard let self = self else { return }
+			let sharesWithReshares = sharesWithReshares.filter { (share) -> Bool in
+				return share.type == .link
+			}
+			self.shares = sharesWithReshares
+			OnMainThread {
+				self.removeShareSections()
+				self.addShareSections()
+				self.handleEmptyShares()
+			}
+		}, keepRunning: true)
+		shareQuery?.refreshInterval = 2
+	}
+
+	override func viewWillAppear(_ animated: Bool) {
+		super.viewWillAppear(animated)
+		navigationController?.isToolbarHidden = true
+	}
+
+	override func viewDidAppear(_ animated: Bool) {
+		super.viewDidAppear(animated)
+		handleEmptyShares()
+	}
+
+	// MARK: - Sharing UI
+
+	func addSectionFor(type: OCShareType, with title: String) {
+		var shareRows: [StaticTableViewRow] = []
+
+		let user = shares.filter { (share) -> Bool in
+			return share.type == type
+		}
+
+		if user.count > 0 {
+			for share in user {
+				if canEdit(share: share) {
+					shareRows.append( StaticTableViewRow(rowWithAction: { [weak self] (_, _) in
+						guard let self = self else { return }
+						let editPublicLinkViewController = PublicLinkEditTableViewController(style: .grouped)
+						editPublicLinkViewController.share = share
+						editPublicLinkViewController.core = self.core
+						editPublicLinkViewController.item = self.item
+						self.navigationController?.pushViewController(editPublicLinkViewController, animated: true)
+					}, title: share.name!, subtitle: share.permissionDescription(), accessoryType: .disclosureIndicator) )
+				} else {
+					shareRows.append( StaticTableViewRow(rowWithAction: nil, title: share.name!, subtitle: share.permissionDescription(), accessoryType: .none) )
+				}
+			}
+		} else {
+			shareRows.append( StaticTableViewRow(buttonWithAction: { [weak self] (_, _) in
+				self?.addPublicLink()
+			}, title: "Create Public Link".localized, style: StaticTableViewRowButtonStyle.plain))
```

Comment:
I understand the idea of only showing the "Create Public Link" button when there are no public links created yet. It took me a while to find the "+" in the upper-left corner, though. In fact, I only recognized it when I already wanted to dismiss the view controller.

So, from a UX perspective, having this "button" always available may be preferable.